StripeFinancialConnections/StripeFinancialConnections/Source/Native/AccountPicker/AccountPickerViewController.swift

```diff
@@ -387,11 +387,52 @@ final class AccountPickerViewController: UIViewController {
                         selectedAccounts = self.dataSource.selectedAccounts
                     }
 
-                    self.delegate?.accountPickerViewController(
-                        self,
-                        didSelectAccounts: selectedAccounts,
-                        nextPane: linkedAccounts.nextPane
-                    )
+                    // for data flows (external_api), where the user is already
+                    // determined to be a Link consumer, we need to take an
+                    // additional step of saving the accounts to Link
+                    if
+                        self.dataSource.manifest.isNetworkingUserFlow == true,
+                        // make sure the current user is a Link consumer
+                        self.dataSource.manifest.accountholderIsLinkConsumer == true,
+                        // make sure this is not a payment flow; in payment flows,
+                        // the AttachLinkedPaymentAccount handles saving to link
+                        !self.dataSource.manifest.shouldAttachLinkedPaymentMethod,
+                        selectedAccounts.count > 0,
+                        let consumerSessionClientSecret = self.dataSource.consumerSessionClientSecret
+                    {
+                        self.dataSource.saveAccountsToLink(
+                            consumerSessionClientSecret: consumerSessionClientSecret
+                        )
+                        .observe { [weak self] result in
+                            guard let self = self else { return }
+
+                            if case .failure(let error) = result {
+                                self.dataSource
+                                    .analyticsClient
+                                    .logUnexpectedError(
+                                        error,
+                                        errorName: "SaveToLinkError",
+                                        pane: .accountPicker
+                                    )
+                            }
+
+                            self.delegate?.accountPickerViewController(
+                                self,
+                                didSelectAccounts: selectedAccounts,
+                                nextPane: .success
+                            )
```

Comment:
note sure if this is what you're saying in the above comment

> (note that this is missing code to set a success screen caption..will be added in different PR)

but as-is this will show success, even in the result == .failure case, right? intentional for now?

Reply:
sorry, the PR "referenced" is this one now:
- https://github.com/stripe/stripe-ios/pull/3459

the note probably was more confusing than helpful, but was there in case one analyzed the web code and saw missing code

> but as-is this will show success, even in the result == .failure case, right? intentional for now?

yes

this is indeed a bit confusing: one way to understand it is that "Financial Connections" could be thought of as having two layers:
1) There's the Financial Connections product that makes sure users can connect their bank accounts
2) The Link product is able to "super charge" this flow by saving bank accounts, and then later allowing consumers to connect their accounts quicker

This particular code does (2), but if it fails, its OK. User will still complete Financial Connections flow, but they won't have a quicker experience next time (...because account failed to save).

[web code of handling save to link error but going to success](https://git.corp.stripe.com/stripe-internal/stripe-js-v3/blob/3ac43f6e/src/linkedAccounts/inner/components/panes/v3/LinkLoginPane/lib/useHandleSaveToLinkError.ts#L48)